src/info/manifest.c

```diff
@@ -383,6 +386,750 @@ manifestNewInternal(void)
+    // are recreated.
+    if (strBeginsWithZ(info->name, PG_PREFIX_PGSQLTMP))
+    {
+        FUNCTION_TEST_RETURN_VOID();
+        return;
+    }
+
+    // Get build data
+    ManifestBuildData buildData = *(ManifestBuildData *)data;
+    unsigned int pgVersion = buildData.manifest->data.pgVersion;
+
+    // Contruct the name used to identify this file/link/path in the manifest
+    strPtr(info->name);
+    strPtr(buildData.manifestParentName);
+    const String *manifestName = strNewFmt("%s/%s", strPtr(buildData.manifestParentName), strPtr(info->name));
+
+    // Skip excluded files/links/paths
+    if (buildData.excludeSingle != NULL && strLstExists(buildData.excludeSingle, manifestName))
+    {
+        LOG_INFO(
+            "exclude '%s/%s' from backup using '%s' exclusion", strPtr(buildData.pgPath), strPtr(info->name),
+            strPtr(strSub(manifestName, sizeof(MANIFEST_TARGET_PGDATA))));
+
+        FUNCTION_TEST_RETURN_VOID();
+        return;
+    }
+
+    // Process storage types
+    switch (info->type)
+    {
+        // Add paths
+        // -------------------------------------------------------------------------------------------------------------------------
+        case storageTypePath:
+        {
+            // There should not be any paths in pg_tblspc
+            if (strEqZ(buildData.manifestParentName, MANIFEST_TARGET_PGDATA "/" MANIFEST_TARGET_PGTBLSPC))
+            {
+                THROW_FMT(
+                    LinkExpectedError, "'%s' is not a symlink - " MANIFEST_TARGET_PGTBLSPC " should contain only symlinks",
+                    strPtr(manifestName));
+            }
+
+            // Add path to manifest
+            ManifestPath path =
+            {
+                .name = manifestName,
+                .mode = info->mode,
+                .user = info->user,
+                .group = info->group,
+            };
+
+            manifestPathAdd(buildData.manifest, &path);
+
+            // Skip excluded path content
+            if (buildData.excludeContent != NULL && strLstExists(buildData.excludeContent, manifestName))
+            {
+                LOG_INFO(
+                    "exclude contents of '%s/%s' from backup using '%s/' exclusion", strPtr(buildData.pgPath), strPtr(info->name),
+                    strPtr(strSub(manifestName, sizeof(MANIFEST_TARGET_PGDATA))));
+
+                FUNCTION_TEST_RETURN_VOID();
+                return;
+            }
+
+            // Skip the contents of these paths if they exist in the base path since they won't be reused after recovery
+            if (strEq(buildData.manifestParentName, MANIFEST_TARGET_PGDATA_STR))
+            {
+                if (strEqZ(info->name, PG_PATH_PGDYNSHMEM) && pgVersion >= PG_VERSION_94)
+                {
+                    FUNCTION_TEST_RETURN_VOID();
+                    return;
+                }
+
+                if (strEqZ(info->name, PG_PATH_PGNOTIFY))
```

Comment:
In perl we check for the version 90 so why not here in C:

Skip pg_notify/* since these files cannot be reused on recovery
        next if $strFile =~ ('^' . MANIFEST_PATH_PGNOTIFY . '\/') && $self->dbVersion() >= PG_VERSION_90;

Reply:
Yeah, missed that one.  I'll add it after review or you can.

Reply:
Code has been changed to (commit 1fe22c3bb51dbb7070d22d8329ee999290ea8560):
`if (strEqZ(info->name, PG_PATH_PGNOTIFY) && pgVersion >= PG_VERSION_90)`